### PR TITLE
docs: Add docs for `solana-test-validator`

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

No docs for `solana-test-validator`

#### Summary of Changes

Add some

### CODE REVIEW

File: docs/src/developing/test-validator.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+fewer restrictions and more configuration options than the public offerings
+provide. This is easily achieved with the `solana-test-validator` binary, which
+starts a local, full-featured, single-node cluster on the developer's workstation.
+
+## Advantages
+- No RPC rate-limits
+- No airdrop limits
+- Direct [on-chain program](on-chain-programs/overview) deployment
+(`--bpl-program ...`)
+- Clone account's from a public cluster, including programs (`--clone ...`)
```
Comment (CriesofCarrots) [inline docs/src/developing/test-validator.md:14]:
```suggestion
- Clone accounts from a public cluster, including programs (`--clone ...`)
```

File: docs/src/developing/test-validator.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
+---
+title: Solana Test Validator
+---
+During early stage development, it is often convenient to target a cluster with
+fewer restrictions and more configuration options than the public offerings
+provide. This is easily achieved with the `solana-test-validator` binary, which
+starts a local, full-featured, single-node cluster on the developer's workstation.
```
Comment (CriesofCarrots) [inline docs/src/developing/test-validator.md:7]: "local" and "on the developer's workstation" seem redundant to me

File: docs/src/developing/test-validator.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+- Direct [on-chain program](on-chain-programs/overview) deployment
+(`--bpl-program ...`)
+- Clone account's from a public cluster, including programs (`--clone ...`)
+- Configurable transaction history (`--limit-ledger-size ...`)
+- Configurable epoch length (`--slots-per-epoch ...`)
+- Jump to an arbitrary slot (`--warp-slot ...`)
+
+## Installation
+The `solana-test-validator` binary ships with the Solana CLI Tool Suite.
+[Install](/cli/install-solana-cli-tools) it before continuing.
```
Comment (CriesofCarrots) [inline docs/src/developing/test-validator.md:21]:
```suggestion
[Install](/cli/install-solana-cli-tools) before continuing.
```

File: docs/src/developing/test-validator.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+```
+solana-test-validator --help
+```
+
+Next start the test validator
+```
+solana-test-validator
+```
+
+By default basic status information is printed while the process is running.
```
Comment (CriesofCarrots) [inline docs/src/developing/test-validator.md:34]:
```suggestion
By default, basic status information is printed while the process is running.
```

File: docs/src/developing/test-validator.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+* The network address of the [Gossip](/validator/gossip#gossip-overview),
+[Transaction Processing Unit](/validator/tpu) and [JSON RPC](clients/jsonrpc-api#json-rpc-api-reference)
+service, respectively
+
+```
+⠈ 00:36:02 | Processed Slot: 5142 | Confirmed Slot: 5142 | Finalized Slot: 5110 | Snapshot Slot: 5100 | Transactions: 5142 | ◎499.974295000
+```
+* Session running time, current slot of the the three block
+[commitment levels](clients/jsonrpc-api#configuring-state-commitment),
+slot height of the last snapshot, trasaction count,
```
Comment (CriesofCarrots) [inline docs/src/developing/test-validator.md:124]:
```suggestion
slot height of the last snapshot, transaction count,
```

File: docs/src/developing/test-validator.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+During early stage development, it is often convenient to target a cluster with
+fewer restrictions and more configuration options than the public offerings
+provide. This is easily achieved with the `solana-test-validator` binary, which
+starts a local, full-featured, single-node cluster on the developer's workstation.
+
+## Advantages
+- No RPC rate-limits
+- No airdrop limits
+- Direct [on-chain program](on-chain-programs/overview) deployment
+(`--bpl-program ...`)
```
Comment (CriesofCarrots) [inline docs/src/developing/test-validator.md:13]:
```suggestion
(`--bpf-program ...`)
```

File: docs/src/developing/test-validator.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+- Clone account's from a public cluster, including programs (`--clone ...`)
+- Configurable transaction history (`--limit-ledger-size ...`)
```
Comment (CriesofCarrots) [inline docs/src/developing/test-validator.md:15]:
```suggestion
- Configurable amount of transaction history retained (`--limit-ledger-size ...`)
```
^ I'm not completely happy with this suggestion, but the original text suggests a lot more interesting control than what actually exists

**Review by CriesofCarrots — COMMENTED** [review]
Thanks for putting this together!
r+ typos and nits

File: docs/src/developing/test-validator.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+provide. This is easily achieved with the `solana-test-validator` binary, which
+starts a local, full-featured, single-node cluster on the developer's workstation.
+
+## Advantages
+- No RPC rate-limits
+- No airdrop limits
+- Direct [on-chain program](on-chain-programs/overview) deployment
+(`--bpf-program ...`)
+- Clone accounts from a public cluster, including programs (`--clone ...`)
+- Configurable transaction history (`--limit-ledger-size ...`)
```
Comment (t-nelson) [inline docs/src/developing/test-validator.md:15]: @CriesofCarrots how about simply...
```suggestion
- Configurable transaction history retention (`--limit-ledger-size ...`)
```

File: docs/src/developing/test-validator.md (hunk 2 above)
Comment (t-nelson) [inline docs/src/developing/test-validator.md:7]:
```suggestion
starts a full-featured, single-node cluster on the developer's workstation.
```

File: docs/src/developing/test-validator.md (hunk 8 above)
Comment (CriesofCarrots) [reply docs/src/developing/test-validator.md:15]: 💯 way better than my suggestion :)

**Comment by t-nelson** [comment]
Ugh... apparently mergify doesn't honor the merge method switch?  I wanted this one to squash :face_with_head_bandage:

**Comment by CriesofCarrots** [comment]
> Ugh... apparently mergify doesn't honor the merge method switch? I wanted this one to squash 🤕

Heh, it sure doesn't: https://github.com/solana-labs/solana/blob/3e0c0abb536f36dd801cdf8af40dac04944ab165/.mergify.yml#L17

**Pull request merged**